qa/vagrant-helpers.rb

```diff
@@ -1,15 +1,56 @@
 # encoding: utf-8
 require "open3"
+require "bundler"
 
 module LogStash
+  class CommandExecutor
+    class CommandError < StandardError; end
+
+    class CommandResponse
+      attr_reader :stdin, :stdout, :stderr, :exitstatus
+
+      def initialize(stdin, stdout, stderr, exitstatus)
+        @stdin = stdin, 
+        @stdout = stdout
+        @stderr = stderr
+        @exitstatus = exitstatus
+      end
+
+      def success?
+        exitstatus == 0
+      end
+    end
+
+    def self.run(cmd)
+      # This block is require to be able to launch a ruby subprocess
+      # that use bundler.
+      Bundler.with_clean_env do
```

Comment:
I'm wondering why this might work here, and not when I tested this same code inside the Rake task, any idea? I always got this same error: see #5176, we should make sure we can call `vagrant` with `rake foo` and with `bundler exec rake foo`